components/hab/src/cli.rs

```diff
@@ -1106,6 +1106,10 @@ pub fn sub_sup_run(_feature_flags: FeatureFlag) -> App<'static, 'static> {
                                                                        the Supervisor startup specified by a package identifier \
                                                                        (ex: core/redis) or filepath to a Habitat Artifact \
                                                                        (ex: /home/core-redis-3.0.7-21120102031201-x86_64-linux.hart).")
+                            // TODO (DM): These flags can eventually be removed.
+                            // See https://github.com/habitat-sh/habitat/issues/7339
+                            (@arg APPLICATION: --application -a +hidden +takes_value requires[ENVIRONMENT] "DEPRECATED")
+                            (@arg ENVIRONMENT: --environment -e +hidden +takes_value requires[APPLICATION] "DEPRECATED")
```

Comment:
Since we're effectively ignoring these now, I don't think they need to require each other anymore.

Along those lines... what happens if they're now provided, but aren't given a value?

Basically, my ideal situation would just be to loosen our constraints as much as possible, accept them if they're there, in any way, and then just ignore them.

Reply:
Good thought. It looks like making the arg be multiple achieves the behavior we want.

Addressed in [20d99a](https://github.com/habitat-sh/habitat/pull/7335/commits/20d99aa36b4dfb14b5f615107863a756b9e18d41)

Reply:
Perfect